src/components/results/securityanalysis/security-analysis-result-utils.ts

```diff
@@ -0,0 +1,351 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React from 'react';
+import {
+    ConstraintsFromContingencyItem,
+    ContingenciesFromConstraintItem,
+    Contingency,
+    LimitViolation,
+    SecurityAnalysisNmkTableRow,
+    subjectLimitViolations,
+} from './security-analysis.type';
+import { IntlShape } from 'react-intl';
+import {
+    ColDef,
+    ICellRendererParams,
+    PostSortRowsParams,
+    ValueFormatterParams,
+    ValueGetterParams,
+} from 'ag-grid-community';
+import { ContingencyCellRenderer } from 'components/spreadsheet/utils/cell-renderers';
+
+const contingencyGetterValues = (params: ValueGetterParams) => {
+    if (params.data?.contingencyId && params.data?.contingencyEquipmentsIds) {
+        return {
+            cellValue: params.data?.contingencyId,
+            tooltipValue: params.data?.contingencyEquipmentsIds.join('\n'),
+        };
+    }
+};
+
+export const computeLoading = (
+    limitViolation: LimitViolation | subjectLimitViolations | Contingency
+): number | undefined => {
+    return (limitViolation.loading =
+        limitViolation.limitType === 'CURRENT'
+            ? (100 * limitViolation.value) /
+              (limitViolation.limit *
+                  (limitViolation as LimitViolation).limitReduction)
+            : undefined);
+};
+
+export const flattenNmKResultsContingencies = (
+    intl: IntlShape,
+    result: ConstraintsFromContingencyItem[] = []
+) => {
+    const rows: SecurityAnalysisNmkTableRow[] = [];
+
+    result?.forEach(
+        ({
+            subjectLimitViolations = [],
+            elements = [],
+            id,
+            status,
+        }: ConstraintsFromContingencyItem) => {
+            if (!!subjectLimitViolations.length) {
+                rows.push({
+                    limit: 0,
+                    limitType: '',
+                    subjectId: '',
+                    value: 0,
```

Comment:
How come `limit` and `value` are set to 0 by default, it wasn't here in the previous implementation

Reply:
yes, limit: 0, limitType: '', subjectId: '', and value: 0, are not necessary. thank you